### DISCUSSION TITLE
日本語バージョンの説明の追加

### DISCUSSION THREAD

**Discussion by Guest126** [cocoa-mhlw/cocoa#270] [General]
## その機能リクエストは何らかの問題に関連しますか / Is your feature request related to a problem?

日本語の Readme を更新します。

## 解決策についてお書きください / Describe the solution you'd like

License. および About Third Party Software. の日本語訳

## その他 / Additional context

#19 および #65 の再提案です。

「ライセンス表記は日本語訳しない」との結論によりcloseされていますが、該当部分はライセンス<del>表記</del>本文ではないため、日本語訳しても差し支えないと考えています。

**Comment by keiji** [comment]
ふたたび日本語の提案と言うことなので、翻訳する範囲についてディスカッションもかねて検討していきましょうか。

　ぼく自身はライセンス表記およびライセンスに関係する表記は日本語に翻訳しなくていいと言うスタンスです。

　ライセンス表記のみならずライセンスに関係する表記の翻訳は、1つのオープンソースソフトウェアが踏み込まないほうが良い——踏み込むべきでない領域——と考えています。

　ぼくは法律の専門家ではありませんが、他の言語の法律や契約文書を日本語に翻訳するのが一筋縄ではいかないと言うことは理解しているつもりです。ある言葉の指し示す範囲や解釈の幅は言語によって異なり、慎重に慎重を期してもやはりトラブルの元となります。

　たとえば今回 "is licensed"が「ライセンスされています」と翻訳されています。英語で"license"は動詞の用法がありますが日本語で「ライセンス」は名詞的な意味合いが強いようにぼくは思うので、より知らない人向けには「許可されています」とした方が、よりわかりやすいかもしれません。けど、それでは『何を許可されているのか』がわかりませんね。『利用が許可されています』とすると、次に『利用』とは何かを定義する必要が出てくるでしょう。

　もちろん、ソフトウェアに関わったことがある人なら「ライセンスされています」で理解できるでしょう。ぼくはそれで理解できます。理解できるという意味では英語で書かれていても理解できるので、翻訳した時のメリットより翻訳したときに招く混乱（デメリット）が大きい。ぼくはそう考えています。

　MPLを作成したMozillaやOpen Source Initiativeなど公式、準公式の日本語訳があればそこを見に行くのが良いですね。その場合は公式、準公式の日本語訳にリンクをするのが良くて、リポジトリ内に日本語訳を転記する必要はないと考えます。

**Comment by keiji** [comment]
> それに習う必要は無いのでしょうか？

MPLの規定で「こうあるべき」と言うものがあるのであれば従うべきと考えます。MPLの規定に沿っていないというご指摘でしょうか。

> これはライセンスについて書かれた文章では無いと判断した方が良いでしょうか。

Chromium Projectsの方に問い合わせるのが適切と考えますが……？

**Comment by keiji** [comment]
> 日本語訳もありますね。

いただいたリンクは

> Firefox は Mozilla Public License の条項下で提供されます。これにより、あなたは Firefox を使用、複製、配布することができます。あなたはまた、Firefox のソースコードを必要に応じて修正することもできます。Mozilla Public License では、あなたが修正したバージョンを配布することも認めています。

の最初の一文ですね。そこだけを訳文として抜き出して使うことは適切ではないとぼくは考えます。

結局のところ「日本語にしてどんなメリットがあるのか」、そのメリットがデメリットを上回ればというところなのですが、何かお考えがあれば提案いただけると嬉しいです。

**Comment by keiji** [comment]
"Japanese reference translations of the OSI approved open source licenses" に MPL 2.0の訳文があるので、これを参考に付けておくことはできそうですね。

https://licenses.opensource.jp/
https://github.com/opensource-jp/licenses/blob/main/MPL-2.0/MPL-2.0.md

**Comment by Guest126** [comment]
当初報告の「該当部分はライセンス表記ではない」との表現は語弊があったので「該当部分はライセンス本文ではない」という表現に修正しました。

**Comment by Guest126** [comment]
@keiji さんのスタンスはわかっていたつもりですが、それはCOCOA開発チームの回答でもあると理解してよろしいのでしょうか。先日、厚生労働省の技術参与の任を受けたと耳にしました。Collaborator としてコミュニティと開発チームの橋渡しを行うという触れ込みだったと思いますが、例えばそれは #19 で「コミュニティで相反する意見がある」ということを開発チームに伝えるのが役割だと思っていました。しかし、@keiji さんに広い裁量権が与えられ、私見に基づいて Issue を close できる／close していたということであれば話は変わってきます。私はまず @keiji さんを説得することを第一目標としなければなりません。

**Comment by Guest126** [comment]
そもそもに勘違いがあるようですが、私はリポジトリ内に MPL 2.0の日本語訳を転記しようと言っているわけではありません。LICENSE は金科玉条であり、おっしゃる通り慎重になるべき部分だからです。

しかし、私が提出したPRは単に README の英文を数行翻訳するだけのものです。その英文は元々 Covid19Radar のコミュニティによって書かれた説明的な文章で、LICENSE の一部ではありません。

**Comment by keiji** [comment]
ぼくがこのIssueでまず検討したいのは「この1行が英語であることでどのような課題に生まれていて、日本語化することでその課題は解決するか」ということです。
Issueの最初に「その機能リクエストは何らかの問題に関連しますか」を記載するように求めているのは、このIssueがそのように運用されているためです。

Guest126さんや他のコントリビューターの方から「こういう課題があるよ。こうすることでもっと良くなるよ」と言う提案をもらって検討して、提案を受け入れたり、もっと詳しく話を聞くのがCollaboratorとしてのぼくの役目です。

ほかのCollaboratorやContributorもこのIssueを見ることができます。だれでも意見も言えます。ぼくを説得（？）することは必要条件ではありません。また、ぼくとしてもこのIssueを検討していきたいという立場には変わりはありません。ぼくの肩書きとスタンスに矛盾はないと考えています。

IssueのDescriptionの項目「その機能リクエストは何らかの問題に関連しますか / Is your feature request related to a problem?」について「この1行が英語であることでどのような課題に生まれていて、日本語化することでその課題は解決するか」みなさんの意見を聞かせてください。

よろしくお願いします。

**Comment by Meiryo7743** [comment]
まず，ここでの話題は，各言語の README ファイルに記載されている COCOA などのライセンス表記についてのものと理解しています。現行の master ブランチにある README ファイルのライセンス表記（[英語](https://github.com/cocoa-mhlw/cocoa/blob/master/README.md#license)，[日本語](https://github.com/cocoa-mhlw/cocoa/blob/master/README.ja.md#%E3%83%A9%E3%82%A4%E3%82%BB%E3%83%B3%E3%82%B9)）は，全ての言語版において英語で記されている状態です。

上記を踏まえたうえで，この issue では，

- ライセンス表記は原文ママでないとその効験を発揮することはない
- 「**ライセンス表記≠ライセンス本文**」であるから，各言語にあわせて翻訳した表記でもライセンスの効果は十分に適用される

という 2 つの考えが上がっています。

ここで一つ疑問が生じました。それは，「ライセンス表記」の原文とは一体何であるか，ということです。

COCOA に適用されている『MPL ライセンス』の本文（原文）を熟読・理解していない状態ですが（すみません），**[ライセンスの本文中](https://github.com/cocoa-mhlw/cocoa/blob/master/LICENSE.md)** に「ライセンス表記は次のように記すよう定める」という規則があるのでしたら，当然それに従うべきでしょう。ただ，そういった定めがないようでしたら，ライセンス表記を翻訳しても差し支えないと考えます。

この issue 及び pull request によって解決できるかもしれない問題を自分なりに考えてみました。各言語の README に応じてライセンス表記を翻訳することで，ある一つの言語のみを理解する人々にとって COCOA のライセンスを把握する手掛かりになる可能性があります。また，README のそこだけが未翻訳のままだと捉え，同様の issue 及び pull request が繰り返し生まれる事態は避けるべきでしょう。その点において，ライセンス表記のローカライズは適切と思われます（ただし，先述した通り，『MPL ライセンス』の定めによります）。

---

自分は法曹の人間ではなく，全くの素人ですが，この issue の内容を興味深いものと感じております。

- MPL ライセンスの本文（原文）
- MPL ライセンスが適用されいる他のプロジェクトでは，どう表記されているのか
- 多言語の README を擁しているプロジェクトでは，ライセンス表記をどう扱っているのか

これらを少し調べてみることにします。

**Comment by keiji** [comment]
ありがとうございます。

> ライセンス表記は原文ママでないとその効験を発揮することはない
> 「ライセンス表記≠ライセンス本文」であるから，各言語にあわせて翻訳した表記でもライセンスの効果は十分に適用される

ぼくはその2つの考え以前に「現在どのような課題があって、表記を日本語に翻訳することでその課題が解決できるのか」がIssueとして示されていないので、それについて意見を求めています。

ここを曖昧にして「日本語にするかしないか」の検討をしても、提案されている手段で目的を達成できるかをだれも評価できません。

目的が曖昧なまま、仮に翻訳すると決めたとして、では「ライセンスされている」という表現は適切か。FireFoxの記載に倣って「COCOAは Mozilla Public License の条項下で提供されます。」としても、これは原文が "Firefox is made available to you under the terms of the Mozilla Public License." で、現在のREADMEの記述とそもそも異なるので、であれば英語の方もあわせて変更するべきなのか、様々な課題が出てきます。

これは典型的な自転車置き場の議論だとぼくは考えていて、Collaboratorとしてはそうならないように、まずは課題の設定を明確にして、提案されている手段がその課題の解決に合致するかどうかを検討していきたいと繰り返しお願いしています。

> 同様の issue 及び pull request が繰り返し生まれる事態は避けるべきでしょう。

Issueの提案者が新しい課題の解決策として、これまで提案した手段を提案するという意味では同様のIssueやPRは何度出ても良いとぼくは考えています。
時間が経てば状況も変わりますし、CollaboratorやContirbutorの知らない情報をその人が知っていることもあると思います。

**Comment by Guest126** [comment]
英語を日本語に翻訳することで日本語話者の理解を助くのは自明であり、その点に疑問を差し挟む余地があるとは想像していませんでした。翻訳した時のメリットより翻訳したときに招く混乱（デメリット）が大きいという話であれば README.ja.md 自体を廃止されてはどうでしょうか。

**Comment by keiji** [comment]
日本語話者が英語のライセンスを見たときに、意味を理解できないという課題があるということでしょうか。

Guest126さんの言う自明は、個人の経験や能力に基づいたものだと理解しています。
自明と言えば、ぼくにとっては現在の"COCOA is licensed..."の文も、GitHubに訪れてライセンスを気にする人はわかるだろうと思っています。

もちろん、ぼくにとっての当たり前が誰かにとってそうではない可能性はありますので、検討していきましょうといったのはそういう理由からです。

README.ja.mdについて、すでに翻訳されているものですね。過去に件の表記を翻訳の対象から外したという事実を示すものとぼくは受け止めています。

**Comment by Guest126** [comment]
> COCOA に適用されている『MPL ライセンス』の本文（原文）を熟読・理解していない状態ですが（すみません），**[ライセンスの本文中](https://github.com/cocoa-mhlw/cocoa/blob/master/LICENSE.md)** に「ライセンス表記は次のように記すよう定める」という規則があるのでしたら，当然それに従うべきでしょう。ただ，そういった定めがないようでしたら，ライセンス表記を翻訳しても差し支えないと考えます。
 
MPLでは「ライセンス表記は次のように記すよう定める」というのではなく「[別途定めるライセンス通知](https://www.mozilla.org/en-US/MPL/2.0/#exhibit-a---source-code-form-license-notice)を添付したコードをMPLの適用対象とみなす」という建付けになっています([MPL v2.0 1.4. “Covered Software”](https://www.mozilla.org/en-US/MPL/2.0/#definitions)より)。なお、現在のCOCOAはそのライセンス通知をどこにも添付していません(Issue #80)。

一例として Mozilla は Firefox Mobile の各ソースコードの最上部と README.md の最下部にそのライセンス通知を記載しています。
https://github.com/mozilla-mobile/fenix/blob/master/README.md

原文を排して日本語に置き換えてしまうのはどうかと思いますが、原文通りのライセンス通知を添付する限り、README.ja.md で追加的に日本語での説明を提供することについては問題ないように思います。

**Comment by keiji** [comment]
ありがとうございます。
"提供することは問題ないように思う"ではなく"現状どのような課題があって、その課題を解決するために日本語訳することが適切"と言う視点で検討をしたいと考えています。ご協力をお願いいたします。

ご指摘のライセンス通知の添付については、"Exhibit A" には続きがありまして、

> If it is not possible or desirable to put the notice in a particular file, then You may include the notice in a location (such as a LICENSE file in a relevant directory) where a recipient would be likely to look for such a notice.

これはリポジトリ管理者が、そちらのほうが望ましいと判断すれば各ソースコードに記載せず、LICENSEに記載するだけも認められていると言う意味であるとぼくは理解しています。

一方、各ファイルにライセンスを記載すること自体は今後、他のライセンスのコードをコピーしてリポジトリに含めることになった時などに、開発者やライセンス管理が混乱することを避ける効果が期待できるので望ましいとぼくは考えています（その点については #80 にも記載したとおりです。

**Comment by Guest126** [comment]
#80 が片付きそうなのでこちらに戻ります。

> 日本語話者が英語のライセンスを見たときに、意味を理解できないという課題があるということでしょうか。

いいえ、「英語のライセンスを見たとき」ではなく「Readme の英語を見たとき」の話です。

> Guest126さんの言う自明は、個人の経験や能力に基づいたものだと理解しています。

いいえ、私が言っているのは一般論です。例えバイリンガルであっても、英語を日本語に翻訳することで日本語話者の理解を助くことに異論を唱える人はいないはずです。

> 自明と言えば、ぼくにとっては現在の"COCOA is licensed..."の文も、GitHubに訪れてライセンスを気にする人はわかるだろうと思っています。

それは「自明」という述語が同じだけの別の話です。対応優先度には影響を与えるでしょうが、対応しない理由にはなりません。

> README.ja.mdについて、すでに翻訳されているものですね。過去に件の表記を翻訳の対象から外したという事実を示すものとぼくは受け止めています。

はい、確かに Covid-19Radar の時点で翻訳の対象から外れていましたが、COCOAとして今どうすべきかを検討するべきだと思います。

**Comment by keiji** [comment]
> いいえ、「英語のライセンスを見たとき」ではなく「Readme の英語を見たとき」の話です。

日本語話者がReadme の英語を見たときに、意味を理解できないという課題があるということでしょうか。

課題を伺った上で、ぼくはReadmeのライセンスに関する英語を日本語に翻訳する理由は乏しいと思っています。

Guest126 さんの仰ることは、ご本人も認めておられるようにまったくの一般的な話に過ぎず、具体的に困っている人やCOCOAとしての課題に結びついていません。
たとえばReadmeを翻訳したとしても、中のプログラムやコメントには英語のものもあるわけです。扱っているもの（ソフトウェア・プログラム）の性質を鑑みずに、一般論を主張されることは適当ではないと考えます。

**Comment by keiji** [comment]
「たった数行の変更をなぜ頑なに受け入れようとしないのか」と疑問に思われる人もいると思うので、補足して少し説明します。

結局のところ「COCOAにどんな課題があるの？」「その課題を解決する目的に提案されている手段は正しい？」ここがマッチして、それをするメリットが、支払うコスト（この場合はヒューマンリソースですね）を上回りそうだなと思わなければ、ぼくとしては「あ、それは今のCOCOAには必要ないですね」といいます。

今回の日本語訳の例でいえば「ソースコードを見に来た人が困ってるんだよ」と言われて、英語が読めない層がリポジトリに訪れていることを教えてもらったり、もっと下世話な話「日本語訳した方がコントリビューター増えるよ！」とかがあれば「やろうやろう」と言いやすいですね。

こうやって「必要ないです」と言うとことについてお叱りを受けていることも承知しています。その上で、ぼくはオープンソースプロジェクトがコントリビューターから提案されたことを受け入れなかったり即Closeすることは普通だと思っていて、むしろオープンソースプロジェクトは使えるリソースが限られていますから「やらないこと」を積極的に決めていくことが必要とも考えています。

以前、TensorFlowにコントリビュートしたときのことですが、ドキュメントに古い記述を見つけて修正するPRを提出すると「一行だけの変更はだめ」と即closeとなったことがあります。そのときは、この部分は皆にとって重要なんだよと説明をしてマージされました（機械的に判断して、食い下がったので確認したのでしょうね。

https://github.com/tensorflow/tensorflow/pull/33643

もちろんCOCOAはTensorFlowと比べて規模も小さいですし、CIコストだってかかりません。けれど、やらないことを決めずにコントリビューターからの提案を全て受け入れられるほどのリソースはまだ持ち合わせていないとぼくは認識しています（good first issueをつけた #57 がまだ残っていることからもおわかりかと思います）。

このIssueについてはcloseせず引き続き意見を募集します。ぼくは自分の意見は述べましたから、新しい何かがあればメンションで呼んでください（ #51 をやります ）。

もちろん他のコラボレーターにも、ここに意見があることは共有しておきます。

**Comment by Takym** [comment]
この issue で取り上げられている README.ja.md の翻訳を行うか行わないかを議論する事より、既にPRが提案されていますので、その訳文内容についてを議論する事の方が有意義だと思いました。

**Comment by keiji** [comment]
このIssueも明日あたりDiscussionに移行しますね。
